### ISSUE TITLE
Quit App on Linux (and windows?) when the last Window is closed

### ISSUE THREAD

**Issue by jens-t** [hql287/Manta#233]
Followup on #228 .

If you close the last window in linux the app runs still in the background. If you then start the app again, the indexeddb is blocked, because of the first instance still running in the background. So the second instance wont work.

You can use `killall manta` to close the instances running in the background, but thats not really a fix. Just a temporary workaround.

I would suggest quitting the app, if the last window is closed on linux. the corresponding pull reuqest is #231 

Only question still remaining is this also a problem on windows? I think at least quitting the app when the last window of the app closes is also the desired behavior for apps on windows. Please Correct me if i'm wrong.

**Comment by hql287** [comment]
I think closing the `mainWindow` should quit the app on both Windows and Linux. Do you feel that's natural? If it does then we probably don't even need to handle 1.2

**Comment by jens-t** [comment]
(a) At first i thought closing the `mainWindow` should quit the app only if the previewWindow is also closed. So you can keep a `previewWindow` open when needed.

(b) But since it is more an overlay working together with the `mainWindow` maybe it is better to quit the app on win/linux only and always when the `mainWindow` is closed, as you suggested. 

so, if we settle on (b), point 2 & 3 of your checklist are already working on all systems. and, as you mentioned, point 1.2 is not needed. But point 1.1 should have following behavior:
[ ] Keep the app always open on macOS
[ ] Quit the app always on linux
[ ] Quit the app always on windows

So, if (b), all we need is
```
if (process.platform != 'darwin') app.quit();
```
in the close event handler of the `mainWindow`, is that right?

**Comment by hql287** [comment]
I guess you're right 😄, but I would write it like this:

```
  mainWindow.on('close', event => {
    if (process.platform === 'darwin') {
      event.preventDefault();
      if (isDev) mainWindow.webContents.closeDevTools();
      mainWindow.hide();
    } else {
      app.quit();
    }
  });
```

Just tested it out on macOS and it works as expected.

**Comment by jens-t** [comment]
Does `app.quit()` close the devtools? I make some tests and change the pull request accordingly.

**Comment by hql287** [comment]
> Does app.quit() close the devtools? 

I don't think it does this by default, [but on the `before-quit` event, the app will destroy all windows instances](https://github.com/hql287/Manta/blob/dev/app.js#L410-L416) and that's enough to quit the app. 

I would double check to see if leaving the `devTool` open would result in any side effect, or maybe just close it explicitly:

```
  mainWindow.on('close', event => {
    if (process.platform === 'darwin') {
      event.preventDefault();
      if (isDev) mainWindow.webContents.closeDevTools();
      mainWindow.hide();
    } else {
      if (isDev) mainWindow.webContents.closeDevTools();
      app.quit();
    }
  });
```

**Comment by hql287** [comment]
Hah, this's related to https://github.com/hql287/Manta/issues/141!

**Comment by jens-t** [comment]
feature merged with #231
